### DISCUSSION TITLE
Overriding details in custom example for unexpected error

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [NRWLDev/fastapi-problem#18] [Q&A] [answered]
https://github.com/NRWLDev/fastapi-problem/blob/main/examples/custom.py

In the custom example for an unexpected error, is it possible to override `details` (="division by zero") to for example, `details`="The server has failed. Please try again later" or is the `title` the best place for a custom message?

**Comment by EdgyEdgemond** [comment, marked as answer]
The simplest way is to add a try/except at the location that the error can occur, and raise a Problem with the details you would like. 

The title is the more appropriate location for the generic message you would like, as the details gives a little context as to how the server failed. To change the title of all unhandled server errors change the ServerError title in that example file.

The other way if you have a lot of locations where this error can occur is to use a custom handler for the error type and convert the error to a Problem with the details you would like. 

https://nrwldev.github.io/fastapi-problem/handlers/

**Reply by EdgyEdgemond** [reply]
The try/except route would usually be preferable, as you can then covert the error to a 400 if the 0 was supplied by the user request informing them to change the value to avoid the error, or raise a 500 like current example if the 0 came from some issue on the server side.

i.e if the user could define a `count` and the code did `1/count` or if the server defined `count` as the number of times something had occured today and it was currently 0.
